From 4a1ed5d1a221bdb1106dbdd1dee23bce79762133 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Wed, 13 Aug 2014 16:37:12 +0800
Subject: [PATCH] fsl_dce: fix calltrace for performance_simple test module

Replace smp_processor_id() with raw_smp_processor_id() to fix following
calltrace:

BUG: using smp_processor_id() in preemptible [00000000] code: insmod/648
caller is .dce_loopback_init+0x2a4/0x10a0 [dce_simple_perf_tester]
CPU: 7 PID: 648 Comm: insmod Tainted: G         C   3.10.38-ltsi-WR6.0.0.0_standard #36
Call Trace:
[c00000000685f800] [c00000000000acc0] .show_stack+0x170/0x290 (unreliable)
[c00000000685f8f0] [c000000000a5dc64] .dump_stack+0x28/0x3c
[c00000000685f960] [c0000000004e278c] .debug_smp_processor_id+0x11c/0x130
[c00000000685f9f0] [8000000000385474] .dce_loopback_init+0x2a4/0x10a0 [dce_simple_perf_tester]
[c00000000685fb10] [c000000000001bc4] .do_one_initcall+0x164/0x1c0
[c00000000685fbc0] [c0000000000e065c] .load_module+0x18bc/0x23a0
[c00000000685fd40] [c0000000000e1300] .SyS_finit_module+0xa0/0xd0
[c00000000685fe30] [c000000000000618] syscall_exit+0x0/0x8c

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 .../tests/performance_simple/dce_perf_simple.c     |    2 +-
 .../tests/performance_simple/dce_sf_perf_simple.c  |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c b/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c
index d6a2cce..54bcfea 100644
--- a/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c
+++ b/drivers/staging/fsl_dce/tests/performance_simple/dce_perf_simple.c
@@ -782,7 +782,7 @@ static int do_operation(void)
 	struct dce_process_cf_gzip_req *def_process_req;
 	struct dce_bman_cfg bcfg;
 
-	pr_info("DCE thread on cpu %d\n", smp_processor_id());
+	pr_info("DCE thread on cpu %d\n", raw_smp_processor_id());
 
 	bcfg.tsize = b_sg_block_size_code;
 	bcfg.tbpid = bman_get_params(pool_sg)->bpid;
diff --git a/drivers/staging/fsl_dce/tests/performance_simple/dce_sf_perf_simple.c b/drivers/staging/fsl_dce/tests/performance_simple/dce_sf_perf_simple.c
index ad6bace..f6ef646 100644
--- a/drivers/staging/fsl_dce/tests/performance_simple/dce_sf_perf_simple.c
+++ b/drivers/staging/fsl_dce/tests/performance_simple/dce_sf_perf_simple.c
@@ -880,7 +880,7 @@ static int do_operation(void)
 	uint32_t total_out = 0;
 	char *p_out;
 
-	pr_info("DCE thread on cpu %d\n", smp_processor_id());
+	pr_info("DCE thread on cpu %d\n", raw_smp_processor_id());
 
 	if (bman_output)
 		pr_info("Using BMan for output\n");
-- 
1.7.5.4

